### PR TITLE
sync: Apply descriptor dynamic offset

### PR DESCRIPTION
**Pull request by artem-lunarg**
Addresses https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/8084 (at least non-overlapping accesses part)

### CODE REVIEW

File: layers/state_tracker/descriptor_sets.cpp [hunk 1]
```diff
@@ -519,6 +519,21 @@ void vvl::DescriptorSet::NotifyInvalidate(const NodeList &invalid_nodes, bool un
     }
 }
 
+uint32_t vvl::DescriptorSet::GetDynamicOffsetIndexFromBinding(uint32_t dynamic_binding) const {
+    const uint32_t index = layout_->GetIndexFromBinding(dynamic_binding);
+    if (index == bindings_.size()) {  // binding not found
+        return vvl::kU32Max;
+    }
+    assert(IsDynamicDescriptor(bindings_[index]->type));
```
Comment (spencer-lunarg) [inline layers/state_tracker/descriptor_sets.cpp:527]: if we are only adding the index from `[0, index - 1]` below, why does this need to be true?

Comment (artem-lunarg) [reply layers/state_tracker/descriptor_sets.cpp:527]: `uint32_t GetDynamicOffsetIndexFromBinding(uint32_t dynamic_binding)` is defined only for dynamic descriptor bindings. Non-dynamic binding is not associated with any dynamic offset. Added a comment.

File: tests/unit/sync_val_positive.cpp [hunk 2]
```diff
@@ -1534,3 +1534,113 @@ TEST_F(PositiveSyncVal, SignalUnsignalSignalSingleSubmit) {
...
+    RETURN_IF_SKIP(InitState());
+
+    // 32 bytes
+    const VkDeviceSize uniform_data_size = 8 * sizeof(uint32_t);
+    // 128 bytes or more (depending on minUniformBufferOffsetAlignment)
+    const VkDeviceSize copy_dst_area_size = std::max((VkDeviceSize)128, m_device->phy().limits_.minUniformBufferOffsetAlignment);
+    // 160 bytes or more (depending on minUniformBufferOffsetAlignment)
+    const VkDeviceSize size = copy_dst_area_size + uniform_data_size;
+
+    // At first we have at least 128 bytes copy destinatin region, followed by 32 bytes of uniform data.
```
Comment (arno-lunarg) [inline tests/unit/sync_val_positive.cpp:1550]:
```suggestion
    // At first we have at least 128 bytes copy destination region, followed by 32 bytes of uniform data.
```

**Review by arno-lunarg — APPROVED** [review]

File: tests/unit/sync_val_positive.cpp (hunk 2 above)
Comment (artem-lunarg) [reply tests/unit/sync_val_positive.cpp]: Updated

**Review by spencer-lunarg — APPROVED** [review]

**Pull request merged**